lib/prettyTitle.js

```diff
@@ -0,0 +1,36 @@
+/*
+ * Given a key name for a message or connector, create a user friendly "title"
+ * parameter for the connectors.json.
+ *
+ * FYI - only ever done when not already declared.
+ */
+const { sentenceCase, unCamelCase } = require('./internal/mout');
+const { UserInputError } = require('./errors');
+
+// Looks for words between word boundaries. Using `gi` so looks for matches globally and case insensitively
+const matchWord = (word) => {
+	return new RegExp(`\\b${word}\\b`, 'gi');
+};
+
+const IdRegex = matchWord('id');
+const IdsRegex = matchWord('ids');
+const UrlRegex = matchWord('url');
+const DdlRegex = matchWord('ddl');
+
+const handleNameEdgeCases = (sentenceCasedTitle) => {
+	return sentenceCasedTitle
+		.replace(IdRegex, 'ID')
+		.replace(IdsRegex, 'IDs')
+		.replace(UrlRegex, 'URL')
+		.replace(DdlRegex, 'DDL');
+};
+
+module.exports = (name) => {
+	if (!name && typeof name !== 'string') {
+		throw new UserInputError(
+			'Please provide a name value in string format.',
+		);
+	}
+	const title = sentenceCase(unCamelCase(name).replace(/_|-/g, ' '));
```

Comment:
Instead of using these 2 functions from mout, you can use the lodash equivalent funcs (startCase and capitalize)
```suggestion
	const title = capitalize(startCase(name).replace(/_|-/g, ' '));
```

Reply:
The reason I didn't do that in falafel was because lodash and mout behaviour can be different. 

So if you're replacing with lodash, make sure the behaviour is what you expect.

Reply:
Looking at the test cases, It covers pretty much all cases we need for a pretty title though

Reply:
Triple checked, all the test cases are passing so I think the switch here should still work for this use-case.